tests/bluetooth/tester/src/l2cap.c

```diff
@@ -389,11 +391,46 @@ static void supported_commands(uint8_t *data, uint16_t len)
 	tester_set_bit(cmds, L2CAP_DISCONNECT);
 	tester_set_bit(cmds, L2CAP_LISTEN);
 	tester_set_bit(cmds, L2CAP_SEND_DATA);
+	tester_set_bit(cmds, L2CAP_CONN_PARAM_UPDATE);
 
 	tester_send(BTP_SERVICE_ID_L2CAP, L2CAP_READ_SUPPORTED_COMMANDS,
 		    CONTROLLER_INDEX, (uint8_t *) rp, sizeof(cmds));
 }
 
+static void conn_param_update(const uint8_t *data, uint16_t len)
+{
+	const struct l2cap_conn_param_update_cmd *cmd = (void *)data;
+	struct bt_le_conn_param param = {
+		.interval_min = sys_le16_to_cpu(cmd->interval_min),
+		.interval_max = sys_le16_to_cpu(cmd->interval_max),
+		.latency = sys_le16_to_cpu(cmd->latency),
+		.timeout = sys_le16_to_cpu(cmd->timeout),
+	};
+	struct bt_conn *conn;
+	uint8_t status = BTP_STATUS_FAILED;
+	int err;
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, (bt_addr_le_t *)cmd);
+	if (!conn) {
+		LOG_ERR("Unknown connection");
+		goto rsp;
+	}
+
+	atomic_set_bit(conn->flags, BT_CONN_SLAVE_PARAM_L2CAP);
```

Comment:
@sjanc  The only difference is that we force L2CAP parameter updates here.

Reply:
conn is opaque type and shall not be used like this directly
also, forcing slave param req like this is invalid (as per spec) if both sides support update over LL

so if you need to test this with external controller use one that doesn't support update over LL (or use old PTS dongle)

Reply:
If I set this flag bit, the host will force updates using L2CAP connection parameters. https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/conn.c#L1387

Reply:
> conn is opaque type and shall not be used like this directly

The same is true elsewhere?

Reply:
> so if you need to test this with external controller use one that doesn't support update over LL (or use old PTS dongle)

Do you mean dual-mode PTS dongle when you say old dongle here? I confirm that Core v4.2 supports the LL connection params update. Refer `Vol 6 Part B Section 2.4.2.1.`